src/interfaces.ts

```diff
@@ -187,18 +188,24 @@ export interface CompilerPlugin {
     afterProgramValidate?: (program: Program) => void;
     beforeProgramTranspile?: (program: Program, entries: TranspileObj[]) => void;
     afterProgramTranspile?: (program: Program, entries: TranspileObj[]) => void;
+    beforeProgramGetCodeActions?: (program: Program, file: BscFile, range: Range, codeActions: CodeAction[]) => void;
+    afterProgramGetCodeActions?: (program: Program, file: BscFile, range: Range, codeActions: CodeAction[]) => void;
+    //scope events
     afterScopeCreate?: (scope: Scope) => void;
     beforeScopeDispose?: (scope: Scope) => void;
     afterScopeDispose?: (scope: Scope) => void;
     beforeScopeValidate?: ValidateHandler;
     afterScopeValidate?: ValidateHandler;
+    onScopeGetCodeActions?: (scope: Scope, file: BscFile, range: Range, diagnostics: BsDiagnostic[], codeActions: CodeAction[]) => void;
```

Comment:
@elsassph what are your thoughts on the name of these new events? I didn't see any benefit in having a `before` and `after`, so i figured `on` would be the pattern for this type of event.

Reply:
Maybe without `Get` like `onScopeCodeActions`?
There definitely are a few `before/after` which feel artificial in general.

Reply:
I kinda like the `get`. all of our other events have a verb in them, it feels out of place to have `onScopeCodeActions` without a verb. 

A weak second argument is that `onScopeCodeActions` is kind of ambiguous as to whether it means "get code actions" or "execute code actions". Not that there is an event for "execute", but I like how "get" makes it more explicit.